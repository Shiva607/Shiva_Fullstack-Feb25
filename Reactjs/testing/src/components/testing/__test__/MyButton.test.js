//! Creating snapshot test for MyButton component using react-test-renderer package

// react-test-renderer is deprecated  Itâ€™s better to use @testing-library/react for testing React components.
/* 
import renderer from 'react-test-renderer';
import MyButton from '../MyButton';

describe('button tests', () => {
	test('button test1', () => {
		const tree = renderer.create(<MyButton label="Submit Me" />).toJSON();
		console.log(tree);
		expect(tree).toMatchSnapshot();
	});
});
 */

//! Creating snapshot test for MyButton component using @testing-library/react package

import React from 'react';
import { render } from '@testing-library/react';
import MyButton from '../MyButton';
describe('button tests', () => {
	test('button test1', () => {
		// asFragment is a function returned as part of the object from the render() call in @testing-library/react so we need call this as object.

		const { asFragment } = render(<MyButton label="Click Me" />);
		expect(asFragment()).toMatchSnapshot();
	});
});
