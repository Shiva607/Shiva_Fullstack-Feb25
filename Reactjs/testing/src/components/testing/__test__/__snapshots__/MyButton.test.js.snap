// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button tests button test1 1`] = `
<DocumentFragment>
  <div>
    <button>
      Click Me
    </button>
  </div>
</DocumentFragment>
`;
